### PR TITLE
[SAIC-376]Correct members for images

### PR DESCRIPTION
**Pull request by mirrorcoder**
Fix bug with incorrect transfer members src to dst

### CODE REVIEW

**Comment by miarmak** [comment]
```
ERROR: test_image_members (tests.test_resource_migration.ResourceMigrationTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/vagrant/CloudFerry/devlab/tests/test_resource_migration.py", line 162, in test_image_members
    filtering_data = self.filtering_utils.filter_images(src_images)
  File "/home/vagrant/CloudFerry/devlab/tests/filtering_utils.py", line 51, in filter_images
    for img in src_data_list[:]:
TypeError: 'generator' object has no attribute '__getitem__'
```

**Comment by miarmak** [comment]
And also, maybe there is a sense to separate these 2 commits in 2 pull requests?

**Comment by mirrorcoder** [comment]
Unit-test for patch(copy from another pull)

**Comment by miarmak** [comment]
It is not unit test. It's functional. And also non-working and not updated with comments from those review. I would suggest staying with 2 different pull requests.

**Comment by mirrorcoder** [comment]
yes, functional. non-working. хз почему

**Comment by roman-verchikov** [comment]
Looks good, +1

File: cloudferrylib/os/image/glance_image.py [hunk 1]
```diff
@@ -404,11 +402,11 @@ def deploy(self, info):
         # on this step we need to create map between source ids and dst ones
         LOG.debug("creating map between source and destination image ids")
         image_ids_map = {}
-        dst_img_checksums = {x.checksum: x.id for x in self.get_image_list()}
+        dst_img_name_checksums = {(x.name, x.checksum): x.id for x in self.get_image_list()}
```
Comment (roman-verchikov) [inline cloudferrylib/os/image/glance_image.py]: pep8 violation: line too long (91>79)

File: cloudferrylib/os/image/glance_image.py [hunk 2]
```diff
@@ -404,11 +402,11 @@ def deploy(self, info):
...
         for image_id_src, gl_image in info['images'].iteritems():
             cur_image = gl_image["image"]
             image_ids_map[cur_image["id"]] = \
-                dst_img_checksums[cur_image["checksum"]]
+                dst_img_name_checksums[(cur_image["name"], cur_image["checksum"])]
```
Comment (roman-verchikov) [inline cloudferrylib/os/image/glance_image.py]: Same as above, pep8 violation

**Comment by roman-verchikov** [comment]
Please fix inlines

**Comment by roman-verchikov** [comment]
Looks good other than pep8 violations inline

**Comment by miarmak** [comment]
Looks good, except of commit message. 
Please, make the first line limited to 50 characters, insert a single blank line after it and also add more detailed commit message below (72 characters max).

**Comment by roman-verchikov** [comment]
@mirrorcoder please update commit message and make sure your changes pass pep8

**Comment by mirrorcoder** [comment]
fix

File: cloudferrylib/os/image/glance_image.py [hunk 3]
```diff
@@ -404,11 +402,14 @@ def deploy(self, info):
         # on this step we need to create map between source ids and dst ones
         LOG.debug("creating map between source and destination image ids")
         image_ids_map = {}
-        dst_img_checksums = {x.checksum: x.id for x in self.get_image_list()}
+        dst_img_name_checksums = {(x.name,
+                                   x.checksum):
+                                      x.id for x in self.get_image_list()}
```
Comment (roman-verchikov) [inline cloudferrylib/os/image/glance_image.py]: pep8 violation:
```
cloudferrylib/os/image/glance_image.py:407:39: E127 continuation line over-indented for visual indent
```
please resolve

**Comment by ntymtsiv** [comment]
tested on devlab, works good. +1 when pep8 violation fixed

**Comment by roman-verchikov** [comment]
@mirrorcoder please resolve unit tests failures and pep8

**Comment by roman-verchikov** [comment]
+1, can be merged

**Pull request merged**